src/parser/Lexer.h

```diff
@@ -322,6 +218,110 @@ extern const char* TemplateOctalLiteral;
+            StringView valueStringLiteralData;
+            double valueNumber;
+            ScanTemplteResult* valueTemplate;
+            ScanRegExpResult valueRegexp;
+            KeywordKind valueKeywordKind;
+        };
+
+        StringView relatedSource();
+        StringView valueStringLiteral();
+        Value valueStringLiteralForAST();
+
+        ~ScannerResult();
+
+        inline void operator delete(void* obj)
+        {
+        }
+
+        inline void operator delete(void*, void*) {}
+        inline void operator delete[](void* obj) {}
+        inline void operator delete[](void*, void*) {}
+        ScannerResult()
+        {
+        }
+
+        ScannerResult(Scanner* scanner, Token type, size_t lineNumber, size_t lineStart, size_t start, size_t end)
+            : valueNumber(0)
+        {
+            this->scanner = scanner;
+            this->type = type;
+            this->startWithZero = this->octal = false;
+            this->hasKeywordButUseString = true;
+            this->plain = false;
+            this->lineNumber = lineNumber;
+            this->lineStart = lineStart;
+            this->start = start;
+            this->end = end;
+        }
+
+        ScannerResult(Scanner* scanner, Token type, const StringView& valueString, size_t lineNumber, size_t lineStart, size_t start, size_t end, bool plain)
+            : valueStringLiteralData(valueString)
+        {
+            this->scanner = scanner;
+            this->type = type;
+            this->startWithZero = this->octal = false;
+            this->hasKeywordButUseString = true;
+            this->plain = plain;
+            this->lineNumber = lineNumber;
+            this->lineStart = lineStart;
+            this->start = start;
+            this->end = end;
+        }
+
+        ScannerResult(Scanner* scanner, Token type, double value, size_t lineNumber, size_t lineStart, size_t start, size_t end)
+            : valueNumber(value)
+        {
+            this->scanner = scanner;
+            this->type = type;
+            this->startWithZero = this->octal = false;
+            this->hasKeywordButUseString = true;
+            this->plain = false;
+            this->valueNumber = value;
+            this->lineNumber = lineNumber;
+            this->lineStart = lineStart;
+            this->start = start;
+            this->end = end;
+        }
+
+        ScannerResult(Scanner* scanner, Token type, ScanTemplteResult* value, size_t lineNumber, size_t lineStart, size_t start, size_t end)
+            : valueTemplate(value)
+        {
+            this->scanner = scanner;
+            this->type = type;
+            this->startWithZero = this->octal = false;
+            this->hasKeywordButUseString = true;
+            this->plain = false;
+            this->lineNumber = lineNumber;
+            this->lineStart = lineStart;
+            this->start = start;
+            this->end = end;
+        }
+
+    private:
+        void consturctStringLiteral();
```

Comment:
`consturct` is a typo. Would you please fix it together? Thank you.

Reply:
Sure, no problem. Patch updated. Thank you for the review.